src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -1006,6 +1006,49 @@ public static void Main()
         }
 
         [Fact]
+        public void TestCallerMemberName_LocalFunction()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+using System;
+
+class D
+{
+    public void LocalFunctionCaller()
+    {
+        void Local()
+        {
+            void LocalNested() => Test.Log();
+            LocalNested();
+        }
+        Local();
+    }
+}
+
+class Test
+{
+    public static int Log([CallerMemberName] string callerName = """")
+    {
+        Console.WriteLine(""name: "" + callerName);
+        return 1;
+    }
+
+    public static void Main()
+    {
+        var d = new D();
+        d.LocalFunctionCaller();
+    }
+}";
+
+            var expected = @"
```

Comment:
Why are we only testing nested local functions here?  Isn't the scenario applicable to top level ones too?